### PR TITLE
JSDocs for activity files

### PR DESCRIPTION
**Pull request by ananyeet**
### What is a quick description of the change?

Write properly formatted jsdocs.

### Is this fixing an issue?

Addresses #94 

### Are there more details that are relevant?

This PR also modifies some syntax. No functional changes.

### Check lists (check `x` in `[ ]` of list items)

- [  ] Test written/updating
- [  ] Tests passing
- [  ] Coding style (indentation, etc)

Please explain why any are not present, if any.
no need, these are jsdocs.

### CODE REVIEW

File: frontend/src/components/ViewActivities/activity.js [hunk 1]
```diff
@@ -80,7 +73,9 @@ class Activity extends React.Component {
           <Accordion.Toggle as={Card.Header} eventKey='0' align='center'>
             {activity[DB.ACTIVITIES_TITLE]}
           </Accordion.Toggle>
-          <Accordion.Collapse eventKey='0' className={'view-activity' + (this.state.editing? ' edit': '')}>
+          <Accordion.Collapse eventKey='0' 
+            className={'view-activity' + (this.state.editing? ' edit': '')}
+            >
```
Comment (zghera) [inline frontend/src/components/ViewActivities/activity.js:78]: Very small nit: something I mentioned in one of the later PRs about the style Keiffer used for this (that I also stole). Up to you if you want to use it.
```suggestion
          <Accordion.Collapse 
            eventKey='0' 
            className={'view-activity' + (this.state.editing? ' edit': '')}
          >
```

File: frontend/src/components/ViewActivities/activityfns.js [hunk 2]
```diff
@@ -43,9 +46,8 @@ export function getField(
 
 /**
  * Sort a list of trip activities by date. 
- * @param {ActivityInfo[]} tripActivities Array of activities.
- * @return {DayOfActivities[]} List of trip activities in the form
- * <pre><code>[ , ...]</code></pre>
+ * @param {!ActivityInfo[]} tripActivities Array of activities.
+ * @return {DayOfActivities[]} List of {@link DayOfActivities}
```
Comment (zghera) [inline frontend/src/components/ViewActivities/activityfns.js:50]: Nullable or no?

File: frontend/src/components/ViewActivities/activityfns.js [hunk 3]
```diff
@@ -90,10 +90,11 @@ export function compareActivities(a, b) {
 /**
  * Write contents into an activity already existing in the database.
  * 
- * @param {string} tripId Database ID of the trip whose actiivty should be modified.
- * @param {string} activityId Database ID of the activity to be modified.
- * @param {Object} newValues Dictionary of the new values in <code>{fieldName: newValue}</code> form
- * @return {boolean} <code>true</code> if the write was successful, <code>false</code> otherwise. 
+ * @param {!string} tripId Database ID of the trip whose actiivty should be modified.
+ * @param {!string} activityId Database ID of the activity to be modified.
+ * @param {!Objects} newValues Dictionary of the new values in `{fieldName: newValue}`
```
Comment (zghera) [inline frontend/src/components/ViewActivities/activityfns.js:95]: Object**s**: Should that 's' be there?

File: frontend/src/components/ViewActivities/activityfns.js [hunk 4]
```diff
@@ -118,9 +119,11 @@ export async function writeActivity(tripId, activityId, newValues) {
 /**
  * Get the value of a reference. 
  * 
- * @param {Reference} ref Reference to get the value of.
- * @param {string} ignoreValue The "null" or "none" value that ref could be.
- * @param {string} defaultValue Value to return if ref.current.value === ignoreValue.
+ * Note: This function breaks if `ref.current` is null. This is intentional.
+ * 
+ * @param {!Reference} ref Reference to get the value of.
```
Comment (zghera) [inline frontend/src/components/ViewActivities/activityfns.js:124]:
```suggestion
 * @param {!React.RefObject} ref Reference to get the value of.
```

File: frontend/src/components/ViewActivities/activitylist.js [hunk 5]
```diff
@@ -8,17 +8,33 @@ import app from '../Firebase';
 const db = app.firestore();
 
 /**
- * Gets the list of activities from the server. 
+ * ReactJS class component for the list of activities. 
  * 
- * @param {string} tripId The trip ID.
- * @return {ActivityInfo[]} The list of trip activities.
+ * @property {Object} props ReactJS props.
+ * @property {string} tripId The tripID.
```
Comment (zghera) [inline frontend/src/components/ViewActivities/activitylist.js:14]: Consistency with JSDoc for props in `activity.js` (choose one way or other)
```suggestion
 * @property {string} props.tripId The tripID.
```

File: frontend/src/components/ViewActivities/editActivity.js [hunk 6]
```diff
@@ -132,9 +135,9 @@ class EditActivity extends React.Component {
    * so when the country changes here, the values in the timezone dropdown
    * change as well.
    *
-   * @param {ref} ref The reference to attach to the dropdown.
-   * @param {ref} tzref The corresponding time zone reference field. 
-   * @param {string} defaultCountry The default country for the dropdown.
+   * @param {!ref} ref The reference to attach to the dropdown.
+   * @param {!ref} tzref The corresponding time zone reference field.
```
Comment (zghera) [inline frontend/src/components/ViewActivities/editActivity.js:139]:
```suggestion
   * @param {!React.RefObject} ref The reference to attach to the dropdown.
   * @param {!React.RefObject} tzref The corresponding time zone reference field. 
```

File: frontend/src/components/ViewActivities/editActivityFormElements.js [hunk 7]
```diff
@@ -10,13 +10,13 @@ const TIMEWIDTH = 2;
...
- * @param {string} formLabel The label of the field for this FormGroup.
- * @param {string} placeHolder The input's placeholder.
- * @param {ref} ref The input's reference.
- * @returns {HTML} A text element form group.
+ * Create a Text element Form Group for the editActivity form. 
+ * 
+ * @param {!string} controlId FormGroup's control ID.
+ * @param {!string} formLabel The label of the field for this FormGroup. 
+ * @param {!string} placeHolder The input's placeholder. 
+ * @param {?ref} ref The input's reference.
```
Comment (zghera) [inline frontend/src/components/ViewActivities/editActivityFormElements.js:18]:
```suggestion
 * @param {?React.RefObject} ref The input's reference.
```

File: frontend/src/components/ViewActivities/editActivityFormElements.js [hunk 8]
```diff
@@ -10,13 +10,13 @@ const TIMEWIDTH = 2;
...
+ * @return {HTML} A text element form group.
```
Comment (zghera) [inline frontend/src/components/ViewActivities/editActivityFormElements.js:19]: I am not sure if this is more accurate or not, but when I return React stuff like this (not vanilla HTML) I use JSX.Element. You can keep this but thought I would mention if you want to use it too.
```suggestion
 * @return {JSX.Element} A text element form group.
```

File: frontend/src/components/ViewActivities/editActivityFormElements.js [hunk 9]
```diff
@@ -32,12 +32,12 @@ export function textElementFormGroup(controlId, formLabel, placeHolder, ref) {
...
- * @param {string} controlId FormGroup's control ID.
- * @param {string} formLabel The label of the field for this FormGroup.
- * @param {string} dropdown The dropdown.
- * @returns {HTML} a location dropdown form group.
+ * Create a Location Dropdown element Form Group for the editActivity form. 
+ * 
+ * @param {!string} controlId FormGroup's control ID.
+ * @param {!string} formLabel The label of the field for this FormGroup. 
+ * @param {!string} dropdown The dropdown. 
+ * @return {HTML} a location dropdown form group.
```
Comment (zghera) [inline frontend/src/components/ViewActivities/editActivityFormElements.js:40]: Same as above.

File: frontend/src/components/ViewActivities/editActivityFormElements.js (hunk 7 above)
Comment (zghera) [reply frontend/src/components/ViewActivities/editActivityFormElements.js:18]: Last time I will do this one too. Keep an eye out for other instances if you want to make the change.

File: frontend/src/components/ViewActivities/editActivityFormElements.js [hunk 10]
```diff
@@ -51,15 +51,15 @@ export function locationElementFormGroup(controlId, formLabel, dropdown) {
...
- * @returns {HTML} A FormGroup for date, time, and timezone.
+ * 
+ * @param {!string} controlId FormGroup's control ID. 
+ * @param {!string} formLabel Label of the field for this FormGroup. 
+ * @param {?ref} dateRef Date's reference. 
+ * @param {!string} dateDefault Default date. 
+ * @param {?ref} timeRef Time's reference. 
+ * @param {!ref} timeDefault Default time. 
+ * @param {!HTML} tzpicker Timezone picker dropdown. 
+ * @return {HTML} A FormGroup for date, time, and timezone.
```
Comment (zghera) [inline frontend/src/components/ViewActivities/editActivityFormElements.js:62]: Same as above.

File: frontend/src/components/ViewActivities/index.js [hunk 11]
```diff
@@ -51,7 +48,8 @@ class ViewActivities extends React.Component {
   }
 
   /**
-   * Complete "Add Activity" operation
+   * Complete "Add Activity" operation.
+   * @param e Form event.
```
Comment (zghera) [inline frontend/src/components/ViewActivities/index.js:52]: Type of param?

File: frontend/src/components/ViewActivities/index.js [hunk 12]
```diff
@@ -70,6 +68,28 @@ class ViewActivities extends React.Component {
...
+      
+  /** @inheritdoc */
+  componentDidMount() {
+    app.firestore()
+        .collection(DB.COLLECTION_TRIPS)
+        .doc(this.tripId)
+        .get()
+        .then(doc => {
+          this.setState({
+            collaborators: doc.get(DB.TRIPS_COLLABORATORS),
```
Comment (zghera) [inline frontend/src/components/ViewActivities/index.js:80]: Just a heads up when merging master into this branch, the constant in get() has changed.

File: frontend/src/components/ViewActivities/activityfns.js (hunk 2 above)
Comment (ananyeet) [reply frontend/src/components/ViewActivities/activityfns.js:50]: oops

File: frontend/src/components/ViewActivities/activitylist.js [hunk 13]
```diff
@@ -8,17 +8,33 @@ import app from '../Firebase';
...
+
+    this.getActivityList = this.getActivityList.bind(this);
+  }
+
+  /**
+   * Gets the list of activities from the server. 
+   * 
+   * This function sets `this.state.days` to the sorted days.
+   * 
+   * @param {!string} tripId The trip ID.
```
Comment (zghera) [inline frontend/src/components/ViewActivities/activitylist.js:30]: Missed one of the non-nullable primitives but I will approve it anyway 😄

File: frontend/src/components/ViewActivities/editActivity.js [hunk 14]
```diff
@@ -95,16 +99,15 @@ class EditActivity extends React.Component {
...
    * reduce scrolling and ensure that the location corresponds to the time zone.
-   *
-   * Tests done manually using UI.
-   *
-   * @param {string} st Either 'start' or 'end' depending on whether the
+   * 
+   * @param {!string} st Either 'start' or 'end' depending on whether the 
    * timezone is for the start or end timezone.
-   * @return {HTML} HTML dropdown item.
+   * @param {!string} defaultTz The default time zone.
```
Comment (zghera) [inline frontend/src/components/ViewActivities/editActivity.js:105]: Same as above.

File: frontend/src/components/ViewActivities/editActivity.js [hunk 15]
```diff
@@ -132,10 +135,10 @@ class EditActivity extends React.Component {
    * so when the country changes here, the values in the timezone dropdown
    * change as well.
    *
-   * @param {ref} ref The reference to attach to the dropdown.
-   * @param {ref} tzref The corresponding time zone reference field. 
-   * @param {string} defaultCountry The default country for the dropdown.
-   * @return {HTML} HTML dropdown of all the countries with timezones.
+   * @param {!React.RefObject} ref The reference to attach to the dropdown.
+   * @param {!React.RefObject} tzref The corresponding time zone reference field. 
+   * @param {!string} defaultCountry The default country for the dropdown.
```
Comment (zghera) [inline frontend/src/components/ViewActivities/editActivity.js:140]: Same as above.

**Comment by ananyeet** [comment]
Received @Hiram-Silvey 's verbal approval to merge since this is the last day of the internship and he was not able to review the PR.

**Pull request merged**